### DISCUSSION TITLE
How to set publisher_confirms ?

### DISCUSSION THREAD

**Discussion by guishake** [gftea/amqprs#105] [Q&A] [answered]
Hello community,

I am a newbie, coming from the python world. I'm used to [aio_pika](https://aio-pika.readthedocs.io) 
I wonder how to set the `publisher_confirms` using amqprs ?

with `aio_pika` we had to set it when opening a channel [doc is here](https://aio-pika.readthedocs.io/en/latest/apidoc.html)

By the way, I don't know if someone in the community is working on the documentation ?

Thanks a lot !

**Comment by gftea** [comment]
https://www.rabbitmq.com/confirms.html#publisher-confirms

See channel method confirms select https://docs.rs/amqprs/latest/amqprs/channel/struct.Channel.html#method.confirm_select

**Reply by guishake** [reply]
OK I see, thanks,

So I just have to do something like this ?
```rust
let channel: amqprs::channel::Channel = connection.open_channel(None).await.unwrap();
    channel
        .register_callback(DefaultChannelCallback)
        .await
        .unwrap();

    channel.confirm_select(ConfirmSelectArguments::new(true)).await.unwrap();
```

**Reply by gftea** [reply, marked as answer]
Yes
